src/main/java/org/fenixedu/academic/util/CurricularPeriodLabelFormatter.java

```diff
@@ -33,30 +33,35 @@
  */
 public class CurricularPeriodLabelFormatter {
 
-    public static String getFullLabelI18N(CurricularPeriod curricularPeriod, boolean abbreviated, Locale locale) {
+    public static String getFullLabelI18N(CurricularPeriod curricularPeriod, Integer term, boolean abbreviated, Locale locale) {
         StringBuilder result = new StringBuilder();
         while (curricularPeriod.getParent() != null) {
-            buildLabel(curricularPeriod, result, abbreviated, locale);
+            buildLabel(curricularPeriod, term, result, abbreviated, locale);
 
             curricularPeriod = curricularPeriod.getParent();
+            term = null;
```

Comment:
why are you setting "term" to null?

Reply:
It's in a loop.
If we don't clear it, after the first time it's printed then it will be repeated for each "level".